aspnetcore/fundamentals/index.md

```diff
@@ -16,97 +16,121 @@ ms.custom: H1Hack27Feb2017
+The `UseStartup` method on `WebHostBuilder` specifies the `Startup` class for your app:
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
+
+[!code-csharp[Main](../getting-started/sample/aspnetcoreapp/Program2x.cs?highlight=15&range=6-17)]
+
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 [!code-csharp[Main](../getting-started/sample/aspnetcoreapp/Program.cs?highlight=7&range=6-17)]
 
-The `Startup` class is where you define the request handling pipeline and where any services needed by the app are configured. The `Startup` class must be public and contain the following methods:
+---
+
+The `Startup` class is where you define the request handling pipeline and where any services needed by the application are configured. The `Startup` class must be public and contain the following methods:
 
 ```csharp
 public class Startup
 {
+    // This method gets called by the runtime. Use this method to add services to the container.
     public void ConfigureServices(IServiceCollection services)
     {
     }
 
+    // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
     public void Configure(IApplicationBuilder app)
     {
     }
 }
 ```
 
-* `ConfigureServices` defines the services (see [Services](#services) below) used by your app (such as the ASP.NET Core MVC framework, Entity Framework Core, Identity, etc.)
+* `ConfigureServices` defines the services (see [Services](#services) below) used by your application (such as the ASP.NET Core MVC, Entity Framework Core, Identity, etc.).
 
-* `Configure` defines the [middleware](middleware.md) in the request pipeline
+* `Configure` defines the [middleware](xref:fundamentals/middleware) in the request pipeline.
 
-For more information, see [Application startup](startup.md).
+For more information, see [Application startup](xref:fundamentals/startup).
 
 ## Services
 
-A service is a component that is intended for common consumption in an application. Services are made available through dependency injection (DI). ASP.NET Core includes a simple built-in inversion of control (IoC) container that supports constructor injection by default. The built-in container can be easily replaced with your container of choice. In addition to its loose coupling benefit, DI makes services available throughout your app. For example, [logging](logging.md) is available throughout your app.
+A service is a component that is intended for common consumption in an application. Services are made available through dependency injection (DI). ASP.NET Core includes a simple, native inversion of control (IoC) container that supports [constructor injection](xref:mvc/controllers/dependency-injection#constructor-injection) by default. The native container can be easily replaced with your container of choice. In addition to its loose coupling benefit, DI makes services available throughout your application. For example, [logging](xref:fundamentals/logging) is available throughout your application.
```

Comment:
~easily~
avoid easy, simple, etc.
In this case it's not that easy.
First use of DI should xref to DI article.